openedx/core/lib/xblock_services/video_config.py

```diff
@@ -0,0 +1,209 @@
+"""
+Video Configuration Service for XBlock runtime.
+
+This service provides video-related configuration and feature flags
+that are specific to the edx-platform implementation
+for the extracted video block in xblocks-contrib repository.
+"""
+
+import logging
+from typing import Optional
+
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.keys import UsageKeyV2
+
+from openedx.core.djangoapps.video_config.models import (
+    CourseYoutubeBlockedFlag,
+    HLSPlaybackEnabledFlag,
+)
+from openedx.core.djangoapps.video_config.toggles import TRANSCRIPT_FEEDBACK
+from openedx.core.djangoapps.video_pipeline.config.waffle import DEPRECATE_YOUTUBE
+from xmodule.exceptions import NotFoundError
+from openedx.core.lib.xblock_services.video_config_utils import (
+    get_public_video_url,
+    is_public_sharing_enabled,
+)
+from organizations.api import get_course_organization
+
+
+log = logging.getLogger(__name__)
+
+
+class VideoConfigService:
+    """
+    Service for providing video-related configuration and feature flags.
+
+    This service abstracts away edx-platform specific functionality
+    that the Video XBlock needs, allowing the Video XBlock to be
+    extracted to a separate repository.
+    """
+
+    def __init__(self, course_id: Optional[CourseKey] = None):
+        """
+        Initialize the VideoConfigService.
+
+        Args:
+            course_id: The course key for course-specific configurations
+        """
+        self.course_id = course_id
+
+    def is_hls_playback_enabled(self, course_id) -> bool:
+        """
+        Check if HLS playback is enabled for the course.
+
+        Arguments:
+            course_id (CourseKey): course id for whom feature will be checked.
+
+        Returns:
+            bool: True if HLS playback is enabled, False otherwise
+        """
+        return HLSPlaybackEnabledFlag.feature_enabled(course_id)
+
+    def is_youtube_deprecated(self, course_id: CourseKey) -> bool:
+        """
+        Check if YouTube is deprecated for the course.
+
+        Args:
+            course_id: The course key
+
+        Returns:
+            bool: True if YouTube is deprecated, False otherwise
+        """
+        return DEPRECATE_YOUTUBE.is_enabled(course_id)
+
+    def is_youtube_blocked_for_course(self, course_id: CourseKey) -> bool:
+        """
+        Check if YouTube is blocked for the course.
+
+        Args:
+            course_id: The course key
+
+        Returns:
+            bool: True if YouTube is blocked, False otherwise
```

Comment:
```suggestion
```
[nit/optional] This is just my personal taste, so up to you: when a simple method's types are fully annotated and there are no "gotchas" that developers need to know related to the args or return value, docstrings like this feel redundant to me, so I usually omit them for the sake brevity.